lib/ruby-push-notifications/mpns/mpns_result.rb

```diff
@@ -0,0 +1,172 @@
+
+module RubyPushNotifications
+  module MPNS
+    # Class that encapsulates the result of a single sent notification to a single
+    # Device URL
+    # (https://msdn.microsoft.com/en-us/library/windows/apps/ff941100%28v=vs.105%29.aspx)
+    class MPNSResult
+      # @return [String]. Receiver MPNS device URL.
+      attr_accessor :device_url
+    end
+
+    # Indicates that the notification was successfully sent to the corresponding
+    # device URL
+    class MPNSResultOK < MPNSResult
+      # @return [String]. The status of the notification received
+      # by the Microsoft Push Notification Service.
+      attr_accessor :notification_status
+
+      # @return [String]. The connection status of the device.
+      attr_accessor :device_connection_status
+
+      # @return [String]. The subscription status.
+      attr_accessor :subscription_status
+
+      def initialize(device_url, headers = nil)
+        @device_url = device_url
+        @notification_status = headers['X-NotificationStatus']
```

Comment:
This constant strings should be constants, to save in memory and to avoid errors mistyping them somewhere.